go/vt/wrangler/traffic_switcher.go

```diff
@@ -995,6 +1018,56 @@ func (ts *trafficSwitcher) changeTableSourceWrites(ctx context.Context, access a
 	})
 }
 
+func (ts *trafficSwitcher) manageTableLocksOnSources(ctx context.Context, action tableLockAction) error {
+	logStrf := "Performing locking action %s on the following tables on source keyspace " +
+		fmt.Sprintf("%v: %v", ts.SourceKeyspaceName(), ts.Tables())
+	if ts.Tables() == nil || len(ts.Tables()) == 0 {
+		return vterrors.Errorf(vtrpc.Code_INTERNAL, "no tables found in the source keyspace %v associated with the %s workflow", ts.SourceKeyspaceName(), ts.WorkflowName())
+	}
+	switch action {
+	case tableLockAcquire:
+		ts.Logger().Infof(logStrf, "ACQUIRE")
+		return ts.ForAllSources(func(source *workflow.MigrationSource) error {
+			var err error
+			primary := source.GetPrimary()
+			if primary == nil {
+				return vterrors.Errorf(vtrpc.Code_INTERNAL, "no primary found for source shard %s", source.GetShard())
+			}
+			tablet := primary.Tablet
+			if err = ts.TabletManagerClient().LockSpecificTables(ctx, tablet, ts.Tables()); err != nil {
+				ts.Logger().Warningf("Error locking tables %v on source tablet %v: %v", ts.Tables(), tablet, err)
+			}
+			return err
+		})
+	case tableLockRelease:
+		ts.Logger().Infof(logStrf, "RELEASE")
+		return ts.ForAllSources(func(source *workflow.MigrationSource) error {
+			var err error
+			primary := source.GetPrimary()
+			if primary == nil {
+				return vterrors.Errorf(vtrpc.Code_INTERNAL, "no primary found for source shard %s", source.GetShard())
+			}
+			tablet := primary.Tablet
+			if err = ts.TabletManagerClient().UnlockTables(ctx, tablet); err != nil {
+				ts.Logger().Warningf("Error unlocking tables on source tablet %v: %v", tablet, err)
+			}
+			return err
+		})
+	case tableLockCheck:
+		ts.Logger().Infof(logStrf, "CHECK")
+		// This will return an error if we still hold the table locks:
+		// ERROR 1100 (HY000): Table 'vreplication' was not locked with LOCK TABLES
+		//query = "select 1 from _vt.vreplication"
+		// If it doesn't return an error then we've somehow lost our session and the
+		// locks along with it
+		//expectError = true
```

Comment:
Should I just remove this for now? I could also add a new RPC for this on the tablet side.